File: Python/Authored/Fish.py
def fish(shoal):
    level = 1
    total = 0
    fishes = sorted(list(shoal))
    for fish in fishes:
        if int(fish) <= level:
            total += (1*int(fish))
            if total >= (level*4):
                total -= (level*4)
                level += 1
    return level

File: CSharp/Authored/Fish.cs
using System;

namespace CodeWars
{
    public class Game
    {
        public int fish(string shoal)
        {
            int level = 1;
            int total = 0;
            char[] t = shoal.ToCharArray();
            int[] fishes = new int[t.Length];
            for(int i= 0; i < t.Length; i++)
            {
                fishes[i] = (int)Char.GetNumericValue(t[i]);
            }
            Array.Sort(fishes);
            for(int i=0; i<fishes.Length; i++)
            {
                int f = fishes[i];
                if (f <= level)
                {
                    total = total + (1 * f);
                    if (total >= (level * 4))
                    {
                        total = total - (level * 4);
                        level++;
                    }
                }
            }
            return level;
        }
    }
}